blinksy/src/leds/lpd8806.rs

```diff
@@ -0,0 +1,87 @@
+use core::iter::repeat_n;
+
+use crate::{
+    color::{ColorCorrection, LinearSrgb, RgbChannels},
+    driver::clocked::ClockedLed,
+    util::component::Component,
+};
+
+pub trait Lpd8806Order {
+    fn reorder(bytes: [u8; 3]) -> [u8; 3];
+}
+
+pub struct OrderGrb;
+pub struct OrderBrg;
+
+impl Lpd8806Order for OrderGrb {
+    #[inline]
+    fn reorder(bytes: [u8; 3]) -> [u8; 3] {
+        RgbChannels::GRB.reorder(bytes)
+    }
+}
+
+impl Lpd8806Order for OrderBrg {
+    #[inline]
+    fn reorder(bytes: [u8; 3]) -> [u8; 3] {
+        RgbChannels::BRG.reorder(bytes)
+    }
+}
+
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct Lpd8806Generic<O: Lpd8806Order>(core::marker::PhantomData<O>);
```

Comment:
I see what you did here with the `Order` generic. I found what I think is a better way, which I'll do as a follow-up pull request. But happy with your approach.